lib/webfuse/adapter/impl/session_manager.c

```diff
@@ -1,31 +1,45 @@
 #include "webfuse/adapter/impl/session_manager.h"
 #include "webfuse/core/util.h"
+#include "webfuse/core/container_of.h"
 #include <stddef.h>
 
 void wf_impl_session_manager_init(
     struct wf_impl_session_manager * manager)
 {
-    wf_impl_session_init(&manager->session, NULL, NULL, NULL, NULL);
+    wf_dlist_init(&manager->sessions);
```

Comment:
Looking at your implementation, i think you optimized the remove operation :)
Acutally you could simply use a single linked list, for the reason, that you use a sequential search for lookup.
Otherwise a map would be the right datastructure.

Reply:
Switched to singly linked list, in order to reduce complexity.